torii/platform/formal.py

```diff
@@ -0,0 +1,189 @@
+
+	# These are not used in formal mode, so we define them here.
+	resources = []
+	connectors = []
+
+
+	toolchain = 'sby'
+	_sby_required_tools = (
+		'yosys',
+		'yosys-abc',
+		'yosys-smtbmc',
+		'sby',
+		'yices'
+	)
+
+	_sby_file_templates = {
+		**TemplatedPlatform.build_script_templates,
+		'{{name}}.il': r'''
+			# {{autogenerated}}
+			{{emit_rtlil()}}
+		''',
+		'{{name}}.debug.v': r'''
+			/* {{autogenerated}} */
+			{{emit_debug_verilog()}}
+		''',
+		'{{name}}.sby': r'''
+			[options]
+			mode {{platform.mode}}
+			expect {{get_override("expect")|default("pass")}}
+			{% if get_override_int("timeout") %}
+			timeout {{get_override_int("timeout")}}
+			{% endif %}
+			multiclock {{get_override("multiclock")|default("off")}}
+			wait {{get_override("wait")|default("off")}}
+			vcd {{get_override("vcd")|default("on")}}
+			vcd_sim {{get_override("vcd_sim")|default("off")}}
+			fst {{get_override("vcd")|default("off")}}
+			aigsmt {{get_override("aigsmt")|default("yices")}}
+			{% if get_override("tbtop") %}
+			tbtop {{get_override("tbtop")}}
+			{% endif %}
+
+			{% if platform.mode in ("bmc", "prove", "cover")  %}
+			depth {{get_override_int("depth")|default("10")}}
+			append {{get_override_int("append")|default("0")}}
+			append_assume {{get_override("append_assume")|default("on")}}
+			{% endif %}
+
+			[engines]
+			{% if platform.mode in ("bmc", "prove") %}
+			smtbmc {{get_override("smtbmc_opts")|options}} {{get_override("solver")|default("yices")}}
+			{% endif %}
+
+			[script]
+			{% for line in get_override_list("script_prelude") %}
+				{{line}}
+			{% endfor %}
+			{% for file in platform.iter_files(".v") -%}
+				read_verilog -formal {{get_override("read_verilog_opts")|options}} {{file}}
+			{% endfor %}
+			{% for file in platform.iter_files(".sv") -%}
+				read_verilog -formal {{get_override("read_verilog_opts")|options}} -sv {{file}}
+			{% endfor %}
+			{% for file in platform.iter_files(".il") -%}
+				read_rtlil {{file}}
+			{% endfor %}
+			read_rtlil {{name}}.il
+			{% for line in get_override_list("script_interim") %}
+				{{line}}
+			{% endfor %}
+			prep -top {{name}}
+			setattr -unset init w:* a:torii.sample_reg %d
+			{% for line in get_override_list("script_epilogue") %}
+				{{line}}
+			{% endfor %}
+
+			[files]
+			{{name}}.il
+			{% for file in platform.iter_files(".v") -%}
+				{{file}}
+			{% endfor %}
+			{% for file in platform.iter_files(".sv") -%}
+				{{file}}
+			{% endfor %}
+			{% for file in platform.iter_files(".il") -%}
+				{{file}}
+			{% endfor %}
+		'''
+	}
+
+	_sby_command_templates = [
+		r'''
+		{{invoke_tool("sby")}}
+			-f
+			-j {{get_override_int("sby_jobs")|default("1")}}
+			-d {{name}}_formal
+			{{get_override("sby_opts")|options}}
+			{{name}}.sby
+		'''
+	]
+
+	def __init__(self, mode: Literal['bmc', 'prove', 'cover', 'live']) -> None:
+		super().__init__()
+
+		self.mode = mode
+
+	@property
+	def _toolchain_env_var(self) -> str:
+		return super()._toolchain_env_var
```

Comment:
Does this have to exist, or, can it just be removed so the superclass version is used (as would happen with out this anyway)?

Reply:
Yeah that can be dropped, don't even recall why it was there